src/nvim/os/env.c

```diff
@@ -410,14 +410,14 @@ static char *remove_tail(char *p, char *pend, char *name)
 /// @param[out] mustfree Ouput parameter for the caller to determine if they are
 ///           responsible for releasing memory. Must be initialized to false
 ///           by the caller.
-char_u *vim_getenv(char_u *name, bool *mustfree)
+char *vim_getenv(char *name, bool *mustfree)
 {
-  char_u      *p;
-  char_u      *pend;
+  char *p;
+  char *pend;
   int vimruntime;
 
 
-  p = (char_u *)os_getenv((char *)name);
+  p = (char *)os_getenv(name);
```

Comment:
Casting `os_getenv` result to `char *` is not correct (breaks `os_getevn` contract to return a string that can't be modified).
Optimal solution would be making `vim_getenv` return value `const char *` too. Please study that. If it complicates too much, then it's better to `xstrdup` `os_getenv` result and work with that. When a function relies on a returned value not being modified, we shouldn't do it.